### DISCUSSION TITLE
How to set tab titles in session config?

### DISCUSSION THREAD

**Discussion by vanarok** [kovidgoyal/kitty#5613] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/6105977/196989089-081aa100-b0ba-4f89-9a00-000e38f49355.png)
I don't understand how to set the tab titles. In the screenshot they are mixed up for some reason.
It should be the other way around.

**Comment by page-down** [comment, marked as answer]
```conf
new_tab YOUR-FIXED-TAB-TITLE
launch --title window-1
launch --title window-2

new_tab SECOND-TAB
```

https://sw.kovidgoyal.net/kitty/overview/#startup-sessions

BTW:
I would suggest that you do not export environment variables in the zsh function, but provide them to kitty or kitty's child processes.
Otherwise, the environment variables in the current shell environment will be changed.

```shell
MY_ENV=... kitty --session ...
```
If you have all windows in the startup session open in the same folder, there is no need to pass in environment variables and `cd` multiple times.
https://sw.kovidgoyal.net/kitty/invocation/#cmdoption-kitty-directory

```shell
kitty -d /path/to/dir --session ...
```

**Reply by vanarok** [reply]
Thank you for such a detailed response and your advice.
